### ISSUE TITLE
Version convention should fit PEP400

### ISSUE THREAD

**Issue by tai271828** [solvcon/pastsolvcon#252]
When working on https://github.com/solvcon/solvcon/issues/249 and https://github.com/solvcon/devenv/issues/98 , I realized that one of the CI failure of `test_from_package` https://github.com/solvcon/devenv/runs/4594808783?check_suite_focus=true is essentially caused by https://github.com/pypa/setuptools/issues/2497 , and raising the following error message in the CI runs:

```
/home/ubuntu/devenv/flavors/solvcon-01/usr/lib/python3.8/site-packages/setuptools/dist.py:507: UserWarning: The versi
on specified ('0.1.4+') is an invalid version, this may not work as expected with newer versions of setuptools, pip, 
and PyPI. Please see PEP 440 for more details. 
```

The inconsistent and unexpected behavior of `python3 setup.py sdist` before/after the release of https://github.com/pypa/setuptools/issues/2497 to make our CI handle the distribution tarball of SOLVCON confused with `SOLVCON-0.1.4-` and `SOLVCON-0.1.4+`.

I would say `0.1.5` is fair since we did not bump the version for many years but the codebase changed a bit : )

## Additional Information
If we looked into the log of the CI failure https://github.com/solvcon/devenv/runs/4594808783?check_suite_focus=true#step:5:48147 , we would see that the build and function test of gas dynamic (Sodtube example) is runnable, but the `test_from_package` step failed. It means the build itself should be good, and the issue happens when making distributing.

**Comment by yungyuc** [comment]
The `+` is used to indicate that it is _not_ a release.  Even if we bump the version to 0.1.5, we will need to immediately check in a change to make it "0.1.5+" and the issue reappears.  I think a more conforming way is to recode the dev version that setuptools is happy.  Perhaps we can adopt part of the numpy versioning like "0.1.4.dev0".  The numpy version works like:

```
$ python3 -c 'import numpy as np; print(np.__version__)'
1.22.0.dev0+679.g261a769dd
```

**Comment by tai271828** [comment]
`.dev` style looks good to me, especially it is used by a popular and powerful project (so people will get the meaning of version convention easily).

**Comment by tai271828** [comment]
Is this version style `1.22.0.dev0+679.g261a769dd` adopted by `conda` rather than a numpy official version style?[1] Both https://github.com/numpy/numpy/tags and https://pypi.org/project/numpy/#history show the oldie but goodie `rc` style. Linux project also uses `rc` style. Maybe we should also consider it as well?

[1] I randomly google and get this https://pypi.anaconda.org/scipy-wheels-nightly/simple/numpy/

**Comment by yungyuc** [comment]
`1.22.0.dev0+679.g261a769dd` is what I got when building numpy from git.  Building (solvcon) from git will be the primary use case for us.

**Comment by tai271828** [comment]
Please note pull request #253 does not fully fixes #249 , and simply one of the step to fix #249 in the first place. If we build SOLVCON build pull request #253 with `devenv` which FIXED the version `1.21.4` of `numpy`, we will resume and pass the CI run with `Ubuntu` [1].

[1] For example, https://github.com/tai271828/devenv/runs/4990806688?check_suite_focus=true

**Comment by yungyuc** [comment]
This issue seems to be resolved by #253.  Please reopen if I am wrong.
